docs/security/tls_setup.md

```diff
@@ -0,0 +1,162 @@
+# TLS support for Percona Everest
+
+The Percona Everest server now supports Transport Layer Security (TLS)  for all incoming connections. TLS **encrypts** client and server communication. This keeps data safe from being **intercepted or changed**. Administrators can configure server certificates and private keys to enable **secure HTTPS access** to the Everest API server. This strengthens security for production environments.
+
+!!! note
+    - When TLS is enabled, the default server port (8080) will only accept `https` traffic, and `http` traffic will be rejected.
+    - We do not support redirects from `http` to `https`.
+    - Self-signed certificates aren't trusted by most browsers, so it's best to use a trusted certificate from a Certificate Authority (CA) in production.
+
+
+
+## TLS setup with Percona Everest
+
+!!! info "Important"
+    This section provides an example using **Helm**. You can also use the provided options with the Percona Everest CLI by using the flag `—helm.set`.
+
+
+### Use Cert-manager (recommended)     
+
+#### Pre-requisites
+
+Ensure that [cert-manager](https://cert-manager.io/docs/) has been deployed in your infrastructure.
+
+#### Set up Percona Everest using cert-manager
+
+Here are the steps to set up the Percona Everest server using cert-manager:
+{.power-number}
+
+1. Install cert-manager:
+
+    ```sh
+    kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.17.0/cert-manager.yaml
+    ```
+
+    For alternative installation methods, refer to the [documentation](https://cert-manager.io/docs/installation/).
+
+2. Create a Self-Signed issuer.
+
+    Use the following YAML configuration:
+
+    ```sh
+    apiVersion: cert-manager.io/v1
+    kind: ClusterIssuer
+    metadata:
+        name: test-selfsigned
+    spec:
+        selfSigned: {}
+    ```
+  
+    This will create a `ClusterIssuer` named `test-selfsigned ` that can issue certificates using a self-signed method, without the need for an external Certificate Authority (CA).
+
+    !!! note
+        Self-signed issuers are not recommended for production use. We are using a self-signed issuer for simplicity. You can configure additional types of issuers; see the [documentation](https://cert-manager.io/docs/configuration/issuers/) for details.
```

Comment:
We need not mention the self-signed cluster issuer.. We can mention in the pre-requisites that the user needs to have a valid issuer/cluster-issuer setup, and you can attach a link to that doc

Reply:
Fixed.